src/components/InputNumber/InputNumber.tsx

```diff
@@ -214,11 +217,11 @@ function InputNumber({
             />
           </div> */}
           {icon && <InputIconContainer icon={icon} />}
-          {error ? (
+          {error && (
```

Comment:
the logic is possibly wrong here, as `error` could be defined and `hideErrorIcon` could be true.
But then. the `<div>` below this line will still get rendered.

Might need to have a variable above like, `const showErrorIcon = error && !hideErrorIcon` and then use that variable instead of `error` on line 220.

I think the same can be applied to all files you've changed.